source/server/worker_impl.cc

```diff
@@ -81,8 +81,20 @@ void WorkerImpl::removeFilterChains(uint64_t listener_tag,
 
 void WorkerImpl::start(GuardDog& guard_dog) {
   ASSERT(!thread_);
-  thread_ =
-      api_.threadFactory().createThread([this, &guard_dog]() -> void { threadRoutine(guard_dog); });
+
+  // In posix, thread names are limited to 15 characters, so contrive to make
+  // sure all interesting data fits there. The naming occurs in
+  // ListenerManagerImpl's constructor: absl::StrCat("worker_", i). Let's say we
+  // have 9999 threads. We'd need, so we need 7 bytes for "worker_", 4 bytes
+  // for the thread index, leaving us 4 bytes left to distinguish between the
+  // two threads used per dispatcher. We'll call this one "dsp:" and the
+  // one allocated in guarddog_impl.cc "dog:".
+  //
+  // TODO(jmarantz): consider refactoring how this naming works so this naming
+  // architecture is centralized, resulting in clearer names.
+  Thread::Options options{absl::StrCat("dsp:", dispatcher_->name())};
```

Comment:
How about `wrk` or `work`?

Reply:
"wrk" is OK, will wind up being "wrk:worker_382" etc. That's probably fine.

I want to follow-up anyway with another refactor to clean up the naming.